### PR TITLE
DEVPROD-9150: Incorporate Pine's Evergreen task into Evergreen

### PR DESCRIPTION
**Pull request by sophiabarness**
DEVPROD-9150

### Description
Pine now has the ability to automatically check whether any changes done to your repo's documentation will be rejected by Pine (and Docusaurus) as a result of bad Markdown. It does this through an Evergreen task that is automatically triggered when a parent patch (i.e a PR in your repo) is created. This will prevent you from merging changes that will cause issues in Pine. However, we do require some setup on your end to enable this functionality for your repo.

Instructions followed here: https://docs.devprod.prod.corp.mongodb.com/pine/onboarding/#incorporating-pines-evergreen-task 

### Testing
This [patch ](https://evergreen.mongodb.com/patch/66b0e254b83406000744f67b?redirect_spruce_users=true)shows the task write-downstream-expansions-for-pine.

### CODE REVIEW

File: self-tests.yml [hunk 1]
```diff
@@ -421,6 +421,13 @@ functions:
           set -o xtrace
           rm evergreen/bin/output.*
           rm evergreen/bin/jstests/*.xml
+  
+  write-repo-name-to-downstream_expansions.yaml-for-pine:
```
Comment (minnakt) [inline self-tests.yml:425]: this seems a bit long / verbose, could we shorten it to something like: `write-downstream-expansions-for-pine`

**Review by minnakt — COMMENTED** [review]
did you test that you can run the task in a patch?

File: self-tests.yml [hunk 2]
```diff
@@ -721,6 +728,12 @@ tasks:
           permissions: public-read
           display_name: swagger.json (latest)
           patchable: false
+  - name: write-and-set-downstream-expansions
```
Comment (bynn) [inline self-tests.yml:731]: I'd make sure this name includes that it's for pine. Something similar to what Minna said above makes sense!

File: self-tests.yml [hunk 3]
```diff
@@ -421,6 +421,13 @@ functions:
           set -o xtrace
           rm evergreen/bin/output.*
           rm evergreen/bin/jstests/*.xml
+  
+  write-repo-name-to-downstream_expansions.yaml-for-pine:
+    - command: shell.exec
+      params: 
+        script: |
+          touch downstream_expansions.yaml
+          echo "pine_repo_name: evergreen" | tee downstream_expansions.yaml
```
Comment (bynn) [inline self-tests.yml:430]: same comment as below, let's mention that its for pine in the yaml name

**Review by bynn — CHANGES_REQUESTED** [review]

File: self-tests.yml (hunk 3 above)
Comment (sophiabarness) [reply self-tests.yml:430]: I'm not sure if changing the yaml name will still make it work (was following Amado's instructions here:  https://docs.devprod.prod.corp.mongodb.com/pine/onboarding/#incorporating-pines-evergreen-task).

Comment (bynn) [reply self-tests.yml:430]: the name of the file won't affect the command as long as you use the same name to call the `downstream_expansion.set` command below

**Review by bynn — CHANGES_REQUESTED** [review]
left a comment and could you also update the description of the PR with the link to the patch you tested on!

File: self-tests.yml [hunk 4]
```diff
@@ -721,6 +728,12 @@ tasks:
           permissions: public-read
           display_name: swagger.json (latest)
           patchable: false
+  - name: write-downstream-expansions-pine
+    commands: 
+      - func: write-downstream-expansions-for-pine
```
Comment (minnakt) [inline self-tests.yml:733]: the name and the function name being near identical seems confusing to me. maybe we should just inline the command and call this task `write-downstream-expansions-for-pine`

**Review by bynn — APPROVED** [review]
I think it looks good but when you link the patch the task wasn't running so I tested it to make sure it would succeed

**Review by minnakt — APPROVED** [review]
Gentle reminder to not to close the ticket yet after merging since the task isn't complete until project settings is also modified!

**Pull request merged**